python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -94,66 +111,65 @@ def auto_materialize_run_tags(self) -> Mapping[str, str]:
         }
 
     @functools.cached_property
-    def previous_tick_requested_or_discarded_asset_partitions(
-        self,
-    ) -> AbstractSet[AssetKeyPartitionKey]:
+    def previous_tick_requested_or_discarded_subset(self) -> AssetSubset:
         """Returns the set of asset partitions that were requested or discarded on the previous tick."""
         if not self.previous_tick_evaluation:
-            return set()
-        return self.previous_tick_evaluation.get_requested_or_discarded_asset_partitions(
+            return self.empty_subset()
+        return self.previous_tick_evaluation.get_requested_or_discarded_subset(
             asset_graph=self.asset_graph
         )
 
     @functools.cached_property
-    def previous_tick_evaluated_asset_partitions(
-        self,
-    ) -> AbstractSet[AssetKeyPartitionKey]:
+    def previous_tick_evaluated_subset(self) -> AssetSubset:
         """Returns the set of asset partitions that were evaluated on the previous tick."""
         if not self.previous_tick_evaluation:
-            return set()
-        return self.previous_tick_evaluation.get_evaluated_asset_partitions(
-            asset_graph=self.asset_graph
-        )
+            return self.empty_subset()
+        return self.previous_tick_evaluation.get_evaluated_subset(asset_graph=self.asset_graph)
 
-    def get_previous_tick_results(self, rule: "AutoMaterializeRule") -> "RuleEvaluationResults":
-        """Returns the results that were calculated for a given rule on the previous tick."""
-        if not self.previous_tick_evaluation:
-            return []
-        return self.previous_tick_evaluation.get_rule_evaluation_results(
-            rule_snapshot=rule.to_snapshot(), asset_graph=self.asset_graph
+    @functools.cached_property
+    def candidate_has_parents_that_have_or_will_update_subset(self) -> AssetSubset:
```

Comment:
feel free to ignore but this name might read more clearly as `has_parents_that_have_or_will_update_candidate_subset`